src/test/java/org/apache/commons/lang3/builder/ReflectionDiffBuilderTest.java

```diff
@@ -184,4 +209,18 @@ public void testGetExcludeFieldNamesWithNullValuesInExcludedFieldNames() {
         assertEquals("charField", excludeFieldNames[0]);
     }
 
+    @Test
+    public void testEqualsObjectDiff() {
+        final TypeTestEntityClass firstObject = new TypeTestEntityClass();
+        firstObject.id = 1L;
+        firstObject.field1 = "a";
+
+        final TypeTestEntityClass secondObject = new TypeTestEntityClass();
+        secondObject.id = 1L;
+        secondObject.field1 = "b";
+
+        final DiffResult list = firstObject.diff(secondObject);
+        assertEquals(1, list.getNumberOfDiffs());
```

Comment:
This test should assert _what_ the difference is; IOW, make assertions on the other properties of the `DiffResult` and `Diff` instances.